transactions.c

```diff
@@ -45,77 +45,103 @@ void copyTransactionToResult(BookTransaction* transaction, BookTransaction** res
         if (current->book_uid == book_uid)  return current;
         current = current->next;
     }
 }
 
-User* findUserFromID(int user_id, User* head)
+// Linker function
+User* findUserFromID(int user_id)
 {
-    User* current = head;
+    User* current = user_list_head;
     while (current)
     {
         if (current->user_id == user_id)    return current;
         current = current->next;
     }
 }
 
-void rentBook(BookTransaction* transaction, Book* book_head, BookCopy* book_copy_head)
+void rentBook(Book* book, User* user)
 {
-    // Find the copy being rented out
-    BookCopy* rented_copy = findBookCopy(transaction->book_uid, book_copy_head);
-    // Check if this book copy is available for rent
-    if (rented_copy->status == inactive)
+    // Check if source book has available copies in stock
+    if (book->in_stock_count < 1)
     {
-        printf("This copy is unavailable. Please choose a different copy of this book!\n");
+        printf("Rental transaction unsuccessful. No copy available in stock for rent\n--------------\n");
         return;
     }
-    else rented_copy->status = inactive; // Update book copy status
-    // Update source book count and status
-    Book* source_book = findSourceBook(rented_copy->ISBN, book_head);
-    source_book->in_stock_count -= 1;
-    if (source_book->in_stock_count < 1)    source_book->status = inactive;
+
+    // Get the first available copy
+    BookCopy* rented_copy = getAvailableCopyOfBook(book);
+
+    // Create and insert transaction
+    BookTransaction* rental = createTransaction(rented_copy->book_uid, getTodaysDate(), user->user_id, open);
+    insertTransaction(rental, &transaction_list);
+
+    // Print transaction status
+    if (rental)
+    {
+        printf("Rental transaction is successful!\n");
+        printf("Customer: %s\nBook title: %s\nBook Copy UID: %i\n", user->full_name, book->title, rented_copy->book_uid);
+        printf("Please return the book by: %i/%i/%i\n", rental->due_date->month, rental->due_date->day, rental->due_date->year);
+        printf("---------------------------\n");
+    }
+    else printf("Rental unsuccessful. Please try again\n");
+
+    // Update source book count and status if needed
+    book->in_stock_count -= 1;
+    if (book->in_stock_count < 1)    book->status = inactive;
+
+    // Update rented copy status
+    rented_copy->status = inactive;
 }
 
-void returnBook(int book_uid, Book* book_head, BookCopy* book_copy_head, BookTransaction* transaction_head)
+void returnBook(int book_uid)
 {
-    //Update Book Copy status after return
-    BookCopy* returned_book_copy = findBookCopy(book_uid, book_copy_head);
-    returned_book_copy->status = active;
-    // Update source book in stock count and status (if needed) after return
-    Book* source_book = findSourceBook(returned_book_copy->ISBN, book_head);
+    // Find and update the returned book copy
+    BookCopy* returned_copy = findBookCopy(book_uid);
+    returned_copy->status = active;
```

Comment:
What if we got returned_copy to be null? I think we should handle it with some condition